src/main/java/org/apache/sysds/runtime/instructions/cp/SpoofCPInstruction.java

```diff
@@ -138,34 +136,6 @@ public boolean isFederated(ExecutionContext ec) {
 	}
 	
 	public boolean isFederated(ExecutionContext ec, FType type) {
```

Comment:
It seems a bit counter-intuitive that a CP instruction has an `isFederated` method. Could we move this to `FedInstructionUtils.checkAndReplaceCP` which I believe is the only place where a call to `isFederated` is needed. Now that you cut the internals of the method to a single line this should be easy.

Reply:
You're right :)
I changed it to directly call the static method _isFederated()_ from _SpoofFEDInstruction_ and introduced the getter method _getInputs()_ in _SpoofCPInstruction_ and _SpoofSPInstruction_ to get the inputs for that call.